### DISCUSSION TITLE
How to pause/resume requests?

### DISCUSSION THREAD

**Discussion by rogererill** [coil-kt/coil#1700] [Q&A]
Hi, I'm in the process of migrating my app from Picasso to Coil, and I wondered if there was an equivalence to Picasso's method to `.pauseTag(tag)`  and `.resumeTag(tag)`.

I have this piece of code:

```
override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
    super.onScrollStateChanged(recyclerView, newState)
    if (newState == RecyclerView.SCROLL_STATE_DRAGGING) {
        picasso.pauseTag(MY_TAG)
    } else {
        picasso.resumeTag(MY_TAG)
    }
}
```

I have read [this](https://github.com/coil-kt/coil/issues/580#issuecomment-727086406) but I wonder if there's a better way nowadays. I'm not really sure how could I access the `Interceptor` to set it as "paused" when creating the request.

Thanks

**Comment by colinrtwhite** [comment]
@rogererill That's still the recommended way to implement pausing + resuming. If you use the singleton `ImageLoader` then you could make the `PauseInterceptor` and `object` else I'd probably inject it with dependency injection.

It might also be useful for your use case, but you can attach metadata to a request using `ImageRequest.Builder.setParameter`. Those can be accessed inside the `Interceptor`.

**Comment by rogererill** [comment]
Hi @colinrtwhite thanks for your response! A slightly modified implementation of the proposed solution did successfully pause the requests, but then I had to add some code that stored the incoming requests that were coming while the app was in this "paused" state so I could resume then later.

It feels a bit hacky but I guess I managed to find a good workaraound. Thanks again!
